.github/workflows/deploy-docker-images.yml

```
@@ -4,78 +4,45 @@ name: Push docker images to GitHub container registry
 
 on:
   push:
-    branches: [main, staging]
-    tags: ["v*.*.*"]
-  pull_request:
-    branches: [main, staging]
+
+env:
+  registry: ghcr.io/dsd-dbs/capella-dockerimages/
+  images: capella/remote capella/readonly capella/ease capella/cli
 
 jobs:
   deploy-docker-images:
     runs-on: ubuntu-latest
     strategy:
-      fail-fast: true
-      max-parallel: 1
-      matrix:
-        include:
-          - image: ghcr.io/dsd-dbs/capella-dockerimages/base
-            context: base
-          - image: ghcr.io/dsd-dbs/capella-dockerimages/capella/base
-            context: capella
-            base-image: ghcr.io/dsd-dbs/capella-dockerimages/base
-            build-args: |
-              BUILD_TYPE=online
-              CAPELLA_VERSION=5.2.0
-          - image: ghcr.io/dsd-dbs/capella-dockerimages/capella/ease
-            context: ease
-            base-image: ghcr.io/dsd-dbs/capella-dockerimages/capella/base
-            build-args: |
-              BUILD_TYPE=online
-    name: build and push ${{ matrix.context }}
+     matrix:
+       capella_version:
+         - "5.0.0"
+         - "5.2.0"
+         - "6.0.0"
+    name: Build image for Capella ${{ matrix.capella_version }}
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v4.3.0
         with:
-          submodules: recursive
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+          python-version: "3.10"
       - name: Login to github container registry
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Extract metadata for Docker
-        id: meta
-        uses: docker/metadata-action@v4
-        with:
-          images: ${{ matrix.image }}
-          tags: |
-            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
-            type=ref,event=branch
-            type=ref,event=pr
-            type=semver,pattern={{version}}
-      - name: Resolve base image
-        # Transform image:tag to BASE_IMAGE=base-image:tag
-        if: ${{ matrix.base-image }}
-        id: base-image
+      - name: Set outputs
+        id: tag
+        run: echo "branch=$(git rev-parse --abbrev-ref HEAD)" > "$GITHUB_OUTPUT"
+      - name: Build
+        run: make ${{ env.images }} CAPELLA_VERSION=${{ matrix.capella_version }} DOCKER_PREFIX=${{ env.registry }} CAPELLA_DOCKERIMAGES_REVISION=${{ steps.tag.outputs.branch }}
```

```suggestion
        run: make ${{ env.images }} PUSH_IMAGES=1 CAPELLA_VERSION=${{ matrix.capella_version }} DOCKER_REGISTRY=${{ env.registry }} CAPELLA_DOCKERIMAGES_REVISION=${{ steps.tag.outputs.branch }}
```

With these changes, you can get rid of lines 44-48 (the pushing of the image). In addition, there is a flag `DOCKER_REGISTRY` that takes care of the registry -> we can replace `DOCKER_PREFIX` with it.

---

I like to run the tests before I push the images.

---

There could be a benefit of pushing directly before running the tests: 
When the test fails, you can pull the image and try to reproduce it with the pushed version, e.g. because the image building is different in the CI.
But let's keep it as it is for now.